hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncClientScanner.java

```diff
@@ -172,6 +172,7 @@ private CompletableFuture<OpenScannerResponse> callOpenScanner(HBaseRpcControlle
       }
       CompletableFuture<OpenScannerResponse> future = new CompletableFuture<>();
       try {
+        controller.setTableName(loc.getRegion().getTable());
```

Comment:
I'm pretty sure this is unnecessary. This `callOpenScanner` ends up getting called via AsyncSingleRequestRpcRetryingCaller, which does `resetCallTimeout()` before calling the callable. So the change you made in that method should handle this case